index.d.ts

```diff
@@ -54,20 +51,27 @@ export interface EventSourceOptions {
   pollingInterval?: number;
 }
 
-export type EventSourceEvent = MessageEvent | OpenEvent | CloseEvent | TimeoutEvent | ErrorEvent | ExceptionEvent;
+type BuiltInEventMap = {
+  'message': MessageEvent,
+  'open': OpenEvent,
+  'close': CloseEvent,
+  'error': ErrorEvent | TimeoutEvent | ExceptionEvent,
+};
+
+type ResolvedEvent<E extends string, T extends EventType<E>> = T extends BuiltInEventType ? BuiltInEventMap[T] : CustomEvent<T>;
```

Comment:
@wojciechkrol Sorry for all the noise. But I have finally figured out a good solution that should make everyone happy 😀. The `ResolvedEvent` type really only needs to have one type parameter instead of two. I would change it to this (also note the added export keyword):

```ts
export type ResolvedEvent<E extends string> = E extends BuiltInEventType ? BuiltInEventMap[E] : CustomEvent<E>;
```

The `EventSourceListener` type would then need to be changed to this:

```ts
export type EventSourceListener<E extends string = never, T extends EventType<E> = EventType<E>> = (
  event: ResolvedEvent<T>
) => void;
```

This way, to handle all built-in event types, one would write:
```ts
const listener: EventSourceListener = (event) => {...}
```

To handle events of all types, both custom and built-in, one would write:
```ts
const listener: EventSourceListener<MyCustomEvents> = (event) => {...}
```

To handle only custom events, one could simply write:
```ts
const listener = (event: ResolvedEvent<MyCustomEvents>) => {...}
```

Likewise, to handle only events of type 'ping':
```ts
const listener = (event: ResolvedEvent<'ping'>) => {...};
```

Reply:
@EmilJunker Makes sense! I've updated the PR and incorporated your suggestions. I've renamed `ResolvedEvent` to `EventSourceEvent`. Thanks for all your effort. For us, the highest priority is maintaining the current library API and not introducing changes to its behavior. `react-native-sse` is quite popular, and if we were to implement changes from https://github.com/binaryminds/react-native-sse/pull/18, it would cause issues for our community.